### ISSUE TITLE
setting details row visible doesn't work initially

### ISSUE THREAD

**Issue by OlliTietavainenVaadin** [vaadin/vaadin-grid-flow#525]
`<vaadin.version>12.0.5</vaadin.version>`
If you call `grid.setDetailsVisible(item, true)` when you're initializing the Grid, it doesn't open the details row(s). Calling the method later (for example, from a button's click listener) opens the details rows as expected.

Sample code that demonstrates the issue:
```java
    Grid<Bean> grid = new Grid<>();
    List<Bean> items = new ArrayList<>();

    public void setFirstAndSecondItemsVisible() {
        grid.setDetailsVisible(items.get(0), true);
        grid.setDetailsVisible(items.get(1), true);
    }

    public MainView() {
        for (int i = 0; i < 5; i++) {
            Bean bean = new Bean(i+"");
            items.add(bean);
        }
        ListDataProvider<Bean> ldp = new ListDataProvider<>(items);
        grid.setDataProvider(ldp);
        Grid.Column<Bean> nameColumn = grid.addColumn(bean -> bean.getName()).setHeader("name");
        grid.setItemDetailsRenderer(new ComponentRenderer<Button, Bean>(item -> new Button(item.getName())));
        setFirstAndSecondItemsVisible(); // should open first and second details rows
        add(grid,
                new Button("click to open details", e -> {
                    setFirstAndSecondItemsVisible(); // same method call as above
                })
        );
    }

    public static class Bean {
        public String getName() {
            return name;
        }
        public void setName(String name) {
            this.name = name;
        }
        private String name;
        public Bean(String name) {
            this.name = name;
        }
    }
 ```

**Comment by eriklumme** [comment]
As  a workaround, one can create a `Composite<Grid>` wrapper with a `@ClientCallable` method that displays the details.

    private class GridWrapper extends Composite<MyGrid> {
        @ClientCallable
        public void init() {
            itemsToDisplay.forEach(item -> getGrid().setDetailsVisible(item, true));
        }

        public EventsGrid getGrid() {
            return getContent();
        }

        @Override
        protected void onAttach(AttachEvent attachEvent) {
            super.onAttach(attachEvent);
            getContent().getElement().callFunction("$server.init");
        }
    }

To have more control over the grid creation, the `Composite` can override `initContent()`.

**Comment by enver-haase** [comment]
Been horribly bitten by it today in 14.0.0.rc3

**Comment by jcgueriaud1** [comment]
Closed by #732
